stl/inc/xutility

```diff
@@ -2005,33 +2005,33 @@ public:
 
     template <class _BidIt2, enable_if_t<_Range_verifiable_v<_BidIt, _BidIt2>, int> = 0>
     friend constexpr void _Verify_range(const reverse_iterator& _First, const reverse_iterator<_BidIt2>& _Last) {
-        _Verify_range(_Last._Get_current(), _First.current); // note reversed parameters
+        _Verify_range(_Last._Get_current(), _First._Current); // note reversed parameters
     }
 
     template <class _BidIt2 = _BidIt, enable_if_t<_Offset_verifiable_v<_BidIt2>, int> = 0>
     constexpr void _Verify_offset(const difference_type _Off) const {
         _STL_VERIFY(_Off != _Min_possible_v<difference_type>, "integer overflow");
-        current._Verify_offset(-_Off);
+        _Current._Verify_offset(-_Off);
     }
 
     template <class _BidIt2 = _BidIt, enable_if_t<_Unwrappable_v<const _BidIt2&>, int> = 0>
     _NODISCARD constexpr reverse_iterator<_Unwrapped_t<const _BidIt2&>> _Unwrapped() const {
-        return static_cast<reverse_iterator<_Unwrapped_t<const _BidIt2&>>>(current._Unwrapped());
+        return static_cast<reverse_iterator<_Unwrapped_t<const _BidIt2&>>>(_Current._Unwrapped());
     }
 
     static constexpr bool _Unwrap_when_unverified = _Do_unwrap_when_unverified_v<_BidIt>;
 
     template <class _Src, enable_if_t<_Wrapped_seekable_v<_BidIt, const _Src&>, int> = 0>
     constexpr void _Seek_to(const reverse_iterator<_Src>& _It) {
-        current._Seek_to(_It.current);
+        _Current._Seek_to(_It._Current);
     }
 
     _NODISCARD constexpr const _BidIt& _Get_current() const noexcept {
-        return current;
+        return _Current;
     }
 
 protected:
-    _BidIt current{};
+    _BidIt _Current{};

```

Comment:
Perhaps surprisingly, `protected reverse_iterator::current` was mandated by C++98 and this remains true in C++20. See WG21-N4861 23.5.1.1 [reverse.iterator]: https://eel.is/c++draft/reverse.iterator . So, the `reverse_iterator` changes need to be reverted. (More occurrences of Standard-mandated `protected` non-`_Ugly` data members like this are `queue`, `priority_queue`, and `stack`'s `c` and `comp`. Back in the 1990s, people apparently thought it was important to derive from Standard types and access their internal data members. :scream_cat:)

Reply:
![ButWhy](https://user-images.githubusercontent.com/3017120/88269401-3f93b800-ccd4-11ea-8b2e-c5e181c5712d.jpg)

Removed the changes to reverse iterator